component/src/main/java/org/wso2/carbon/identity/authenticator/nuxeo/NuxeoAuthenticator.java

```diff
@@ -0,0 +1,404 @@
+        callbackUrl.setDisplayName(NuxeoAuthenticatorConstants.CALLBACK_URL);
+        callbackUrl.setName(IdentityApplicationConstants.OAuth2.CALLBACK_URL);
+        callbackUrl.setDescription("Enter the callback url");
+        callbackUrl.setDisplayOrder(2);
+        configProperties.add(callbackUrl);
+
+        Property nuxeoServerUrl = new Property();
+        nuxeoServerUrl.setDisplayName(NuxeoAuthenticatorConstants.NUXEO_SERVER_URL);
+        nuxeoServerUrl.setName(NuxeoAuthenticatorConstants.NUXEO_SERVER_URL);
+        nuxeoServerUrl.setDescription("Enter the Nuxeo server url. Eg: http://localhost:8080");
+        nuxeoServerUrl.setDisplayOrder(3);
+        configProperties.add(nuxeoServerUrl);
+
+        return configProperties;
+    }
+
+    /**
+     * This method is overridden to check additional condition like whether request is having
+     * sendToken, token parameters, generateNuxeoToken and authentication name.
+     *
+     * @param request  Http servlet request
+     * @param response Http servlet response
+     * @param context  AuthenticationContext
+     * @throws AuthenticationFailedException User tenant domain mismatch
+     */
+    @Override
+    protected void processAuthenticationResponse(HttpServletRequest request, HttpServletResponse response,
+                                                 AuthenticationContext context) throws AuthenticationFailedException {
+
+        String clientId = null;
+        try {
+            Map<String, String> authenticatorProperties = context.getAuthenticatorProperties();
+            clientId = authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_ID);
+            String clientSecret = authenticatorProperties.get(OIDCAuthenticatorConstants.CLIENT_SECRET);
+            String tokenEndPoint = getTokenEndpoint(authenticatorProperties);
+            String callbackUrl = getCallbackUrl(authenticatorProperties);
+            String userInfoUrl = getUserInfoEndpoint(authenticatorProperties);
+            OAuthAuthzResponse authorizationResponse = OAuthAuthzResponse.oauthCodeAuthzResponse(request);
+            String code = authorizationResponse.getCode();
+            AuthenticatedUser authenticatedUserObj;
+            Map<ClaimMapping, String> claims;
+            OAuthClientRequest accessRequest = getAccessRequest(tokenEndPoint, clientId, code, clientSecret,
+                    callbackUrl);
+            OAuthClient oAuthClient = new OAuthClient(new URLConnectionClient());
+            OAuthClientResponse oAuthResponse = getResponseOauth(oAuthClient, accessRequest);
+            String accessToken = oAuthResponse.getParam(OIDCAuthenticatorConstants.ACCESS_TOKEN);
+            String loggedInUserInfo = getUserInfoJson(userInfoUrl, accessToken);
+            claims = mapUserInfo(loggedInUserInfo);
+            authenticatedUserObj = AuthenticatedUser.createFederateAuthenticatedUserFromSubjectIdentifier(
+                    String.valueOf(JSONUtils.parseJSON(loggedInUserInfo).
```

Comment:
Without parsing `loggedInUserInfo` multiple times, parse once and assign it to a variable